planner/implementation/simple_plans.go

```diff
@@ -36,3 +37,35 @@ type ShowImpl struct {
 func NewShowImpl(show *plannercore.PhysicalShow) *ShowImpl {
 	return &ShowImpl{baseImpl: baseImpl{plan: show}}
 }
+
+// TiDBSelectionImpl is the implementation of PhysicalSelection in TiDB layer.
+type TiDBSelectionImpl struct {
+	baseImpl
+}
+
+// CalcCost implements Implementation CalcCost interface.
+func (sel *TiDBSelectionImpl) CalcCost(outCount float64, childCosts []float64, children ...*memo.Group) float64 {
+	sel.cost = outCount*sel.plan.SCtx().GetSessionVars().CPUFactor + childCosts[0]
```

Comment:
We should use row count of child group here instead of `outCount`?

Reply:
If so, how do we make use of the `ExpectedCnt`? How about using `outCount / selectivity`?

selectivity = sel.plan.stats.RowCount / childGroup.Stats.RowCount

Reply:
Oh, it seems we didn't update the stats of the child group. If we scale the stats of child group by `ExpectedCnt`, we can use the row count of the child group then.